modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java

```diff
@@ -0,0 +1,228 @@
+        accessToken = null;
+    }
+
+    @Test(groups = "wso2.is", description = "Check Oauth2 application registration")
+    public void testRegisterApplication() throws Exception {
+
+        OAuthConsumerAppDTO appDto = createApplication();
+        Assert.assertNotNull(appDto, "Application creation failed.");
+
+        consumerKey = appDto.getOauthConsumerKey();
+        Assert.assertNotNull(consumerKey, "Application creation failed.");
+
+        consumerSecret = appDto.getOauthConsumerSecret();
+        Assert.assertNotNull(consumerSecret, "Application creation failed.");
+    }
+
+    @Test(groups = "wso2.is", description = "Send authorize user request and get access token", dependsOnMethods = "testRegisterApplication")
+    public void testGetAccessToken() throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<>();urlParameters.add(new BasicNameValuePair("grantType",
+                OAuth2Constant.OAUTH2_GRANT_TYPE_CLIENT_CREDENTIALS));
+        urlParameters.add(new BasicNameValuePair("consumerKey", consumerKey));
+        urlParameters.add(new BasicNameValuePair("consumerSecret", consumerSecret));
+        urlParameters.add(new BasicNameValuePair("accessEndpoint",
+                OAuth2Constant.ACCESS_TOKEN_ENDPOINT));
+        urlParameters.add(new BasicNameValuePair("authorize", OAuth2Constant.AUTHORIZE_PARAM));
+        urlParameters.add(new BasicNameValuePair("scope", SYSTEM_SCOPE));
+        HttpResponse response =
+                sendPostRequestWithParameters(client, urlParameters,
+                        OAuth2Constant.AUTHORIZED_USER_URL);
+        Assert.assertNotNull(response, "Authorization request failed. Authorized response is null");
+        EntityUtils.consume(response.getEntity());
+
+        response = sendPostRequest(client, OAuth2Constant.AUTHORIZED_URL);
+
+        Map<String, Integer> keyPositionMap = new HashMap<String, Integer>(1);
+        keyPositionMap.put("name=\"accessToken\"", 1);
+
+        List<KeyValue> keyValues =
+                DataExtractUtil.extractInputValueFromResponse(response,
+                        keyPositionMap);
+        Assert.assertNotNull(keyValues, "Access token Key value is null.");
+        accessToken = keyValues.get(0).getValue();
+
+        EntityUtils.consume(response.getEntity());
+        Assert.assertNotNull(accessToken, "Access token is null.");
+    }
+
+    @Test(groups = "wso2.is", description = "Test introspection endpoint", dependsOnMethods = "testGetAccessToken")
+    public void testIntrospectionEndpoint() throws Exception {
+
+        String scope = getScopesFromIntrospectionResponse();
+        doTheScopeValidationBasedOnTheTestUserMode(scope);
+    }
+
+    @Test(groups = "wso2.is", description = "Test introspection endpoint", dependsOnMethods = "testIntrospectionEndpoint")
+    public void getTokenAndValidate() throws Exception {
+
+        client = HttpClientBuilder.create().disableRedirectHandling().build();
```

Comment:
Is this protected by the finally block?

Reply:
No, I'll move this to the try block as well

Reply:
Fixed with https://github.com/wso2/product-is/pull/15437/commits/1a78966577c03d26c950091c0c927c6bdc183a54